s4e-backend/src/main/java/pl/cyfronet/s4e/data/repository/SceneRepositoryImpl.java

```diff
@@ -0,0 +1,352 @@
+
+    }
+
+    private String getOrderByField(Map<String, Object> params) {
+        switch (params.get("sortBy").toString()) {
+            case "ingestion_time":
+                return "ingestion_time";
+            case "sensing_time":
+                return "sensing_time";
+            default:
+                return "id";
+        }
+    }
+
+    private String getOrderBy(Map<String, Object> params) {
+        return params.get("orderBy").equals("ASC") ? "ASC" : "DESC";
+    }
+
+    private void queryByTime(Map<String, Object> params, StringBuilder resultQuery) {
+        queryBySensingTime(params, resultQuery);
+        queryByIngestionTime(params, resultQuery);
+    }
+
+    private void queryByText(Map<String, Object> params, StringBuilder resultQuery) {
+        queryBySatellitePlatform(params, resultQuery);
+        queryByProductType(params, resultQuery);
+        queryByPolarisation(params, resultQuery);
+        queryByProcessingLevel(params, resultQuery);
+        queryBySensorMode(params, resultQuery);
+        queryByCollection(params, resultQuery);
+        queryByTimeliness(params, resultQuery);
+        queryByInstrument(params, resultQuery);
+        queryByProductLevel(params, resultQuery);
+        queryByRelativeOrbitNumber(params, resultQuery);
+        queryByAbsoluteOrbitNumber(params, resultQuery);
+    }
+
+    private void queryByNumber(Map<String, Object> params, StringBuilder resultQuery) {
+        queryByCloudCover(params, resultQuery);
+    }
+
+    private void timeParameters(PreparedStatement ps, Map<String, Object> params) throws SQLException {
+        parametersBySensingTime(params, ps);
+        parametersByIngestionTime(params, ps);
+    }
+
+    private void textParameters(PreparedStatement ps, Map<String, Object> params) throws SQLException {
+        parametersBySatellitePlatform(params, ps);
+        parametersByProductType(params, ps);
+        parametersByPolarisation(params, ps);
+        parametersByProcessingLevel(params, ps);
+
+        parametersBySensorMode(params, ps);
+        parametersByCollection(params, ps);
+        parametersByTimeliness(params, ps);
+        parametersByInstrument(params, ps);
+        parametersByProductLevel(params, ps);
+        parametersByRelativeOrbitNumber(params, ps);
+        parametersByAbsoluteOrbitNumber(params, ps);
+    }
+
+    private void numberParameters(PreparedStatement ps, Map<String, Object> params) throws SQLException {
+        //float
+        parametersByCloudCover(params, ps);
+    }
+
+    private void queryBySensingTime(Map<String, Object> params, StringBuilder resultQuery) {
+        if (params.containsKey("sensingFrom") && params.containsKey("sensingTo")) {
+            // [TIMESTAMP] sensingFrom / sensingTo -> sensing_time
+            resultQuery.append(" AND to_timestamp(metadata_content->>'sensing_time', 'YYYY-MM-DD\"T\"HH24:MI:SS\"Z\"')");
```

Comment:
- Copy pasted time format: `'YYYY-MM-DD\"T\"HH24:MI:SS\"Z\"'`, shouldn't be stored in const?
- Whole line of timestamp conversion is duplicated: `to_timestamp(metadata_content->>'sensing_time', 'YYYY-MM-DD\"T\"HH24:MI:SS\"Z\"')`